### DISCUSSION TITLE
Redirection system not have a header notation

### DISCUSSION THREAD

**Discussion by Ammarnassar** [livewire/livewire#4343] [3. Feature ideas]
When I use a redirect method with header notation , like this : 

`return redirect(route('login'))->header('refresh', '3');` 

or 

`return redirect(route('login'))->withHeaders(['refresh' => '3']);`

I get the following error : 

**BadMethodCallException**
Method Livewire\Redirector::header does not exist.

**Comment by joshhanley** [comment]
@Ammarnassar it's possible that was never implemented with Livewire's custom redirector.

I've changed this to a feature request and PR's for this would be welcome! 🙂

**Reply by telkins** [reply]
Just for easier navigation, the feature request appears to be #4343.

**Comment by telkins** [comment]
Likely related to #4905 ...  🤓

**Comment by LuanHimmlisch** [comment]
I can't figure out why Livewire has to hijack the redirector with their own that has fewer functions. It's just a headache to work with

**Reply by joshhanley** [reply]
@LuanHimmlisch the reason is that Livewire's redirects need to happen in the browser, as they are AJAX requests, where as a Laravel redirect happens on the backend.

If the custom implementation didn't exist, Laravel would do the redirect on the backend and return a non-Livewire response back to the browser, which then wouldn't be able to be handled and would break Livewire, which to a user would look like nothing happened at all, then after that nothing would work.

We're open to PR's that add any required functionality though.

Hope this helps!

**Reply by LuanHimmlisch** [reply]
@joshhanley you're right, I didn't take that into consideration. Thanks

**Reply by joshhanley** [reply]
@LuanHimmlisch no worries! 🙂
